File: reverse link li in group of size k.cpp
class Solution
{
    public:
    struct node *reverse (struct node *head, int k)
    { 
        // Complete this method
        if (head == nullptr) {
            return head;
        }
        int count = 0;
        node* prev = nullptr;
        node* current = head;
        node* next = nullptr;
        while (current && count < k) {
            next = current->next;
            current->next = prev;
            prev = current;
            current = next;
            count++;
        }
        node* next_head = reverse(next, k);
        head->next = next_head;
        return prev;
        
    }
};

File: reverse link li in group of size k.py
class Solution:
    def reverse(self,head, k):
        if head is None:
            return head
        prev = None
        next = None
        current = head
        count = 0
        while current and count < k:
            next = current.next
            current.next = prev
            prev = current
            current = next
            count += 1
        next_head = self.reverse(next, k)
        head.next = next_head
        return prev
